meteor-app/imports/ui/Report_Builder/Report_Builder.tsx

```diff
@@ -283,13 +304,32 @@ export const Report_Builder = () => {
 				<Input placeholder={'Enter Report Name'} label={'Report Name'} value={reportStructure.name} flex={'flex'}
 					onChange={(e) => handleReportName(e.target.value)}
 				/>
-				
-				<div className="ml-4 ">
+
+
+				<div className="ml-4">
 					<Button onClick={() => createNewTable('static')} text="+ New Static Table" color="green" />
 					<Button onClick={() => createNewTable('collection')} text="+ New Collection Table" color="green" />
+					<Button onClick={() => {
+						Meteor.call('Get_Tags', (error, result) => map_tags(error, result)); setReportStructure(prevState => {
```

Comment:
You should consider moving this block, within `map_tags`

makes the Jsx more readable, and keeps state updating logic, within your declared functions / handlers